components/create-collective/NewCreateCollective.js

```diff
@@ -0,0 +1,223 @@
+    }
+    return;
+  }
+
+  componentDidUpdate(oldProps) {
+    const { query } = this.props;
+    if (oldProps.query.step !== query.step) {
+      if (query.step === 'form') {
+        this.setState({ form: true });
+      } else {
+        this.setState({ form: false });
+      }
+    }
+    if (oldProps.query.category !== query.category) {
+      if (query.category === 'opensource' || query.token) {
+        this.setState({ category: 'opensource' });
+      } else if (query.category === 'community') {
+        this.setState({ category: 'community' });
+      } else if (query.category === 'climate') {
+        this.setState({ category: 'climate' });
+      } else if (!query.category) {
+        this.setState({ category: null });
+      }
+    }
+    return;
+  }
+
+  handleChange(key, value) {
+    this.setState({
+      [key]: value,
+    });
+  }
+
+  async createCollective(CreateCollectiveInputType) {
+    // check we have agreed to TOS
+    if (!CreateCollectiveInputType.tos) {
+      this.setState({
+        error: 'Please accept the terms of service',
+      });
+      return;
+    }
+
+    // set state to loading
+    this.setState({ status: 'loading' });
+
+    // prepare object
+    CreateCollectiveInputType.tags = this.state.category;
+    if (this.state.github) {
+      CreateCollectiveInputType.githubHandle = this.state.github.handle;
+      this.host = { slug: 'opensource' };
+    }
+    delete CreateCollectiveInputType.tos;
+
+    // try mutation
+    try {
+      const res = await this.props.createCollectiveV2({
+        collective: CreateCollectiveInputType,
+        host: this.host ? { slug: this.host.slug } : null,
+        automateApprovalWithGithub: CreateCollectiveInputType.githubHandle ? true : false,
+      });
+      const collective = res.data.createCollective;
+      const successParams = {
+        slug: collective.slug,
+      };
+      this.setState({
+        status: 'idle',
+        result: { success: 'Collective created successfully' },
+      });
+      await this.props.refetchLoggedInUser();
+      if (CreateCollectiveInputType.HostCollectiveId) {
+        successParams.status = 'collectiveCreated';
+        successParams.CollectiveId = collective.id;
+        successParams.collectiveSlug = collective.slug;
+        Router.pushRoute('collective', {
+          slug: collective.slug,
+          status: 'collectiveCreated',
+          CollectiveId: collective.id,
+          CollectiveSlug: collective.slug,
+        });
+      } else {
+        Router.pushRoute('collective', { slug: collective.slug });
+      }
+    } catch (err) {
+      const errorMsg = getErrorFromGraphqlException(err).message;
+      this.setState({ status: 'idle', error: errorMsg });
+    }
+  }
+
+  render() {
+    const { LoggedInUser, query } = this.props;
+    const { category, form, error } = this.state;
+    const { token } = query;
+
+    const canApply = get(this.host, 'settings.apply') || true;
```

Comment:
Better rely on the GraphQL `canApply` property.

Reply:
added